godot/scripts/utils/PlayerVariables.gd

```diff
@@ -1,13 +1,3 @@
 extends Node
 
-var _max_health: int = 3;
-var _health: int = 3;
-
-func get_health():
-	return _health;
-
-func add_health():
-	_health = min(_max_health, _health + 1)
-
-func remove_health():
-	_health = max(0, _health - 1)
+var health: int;
```

Comment:
a bit annoying that I need this autoload just for 1 variable to sync player health value with the UI... any suggestions to factor this out? (or is it just fine?)

Reply:
(I guess the other option is the SignalBus and hooking up the health signal to it)

Reply:
Yeah, we definitely don't need it like this. I would suggest making the health component generic and getting rid of this file. Then, the player script will be responsible for decrementing player health and sending a "PlayerHealthUpdate" signal to the signal bus. Nothing should be directly wired to UI. Instead, UI will only ever listen for signals and update based on those signals.

Reply:
How would you initialize it? Emitting the signal from player's `_ready` doesn't really work as expected. I'm guessing the UI node loads & connects to the signal too late, and never catches the first one (containing the initial health val)

Reply:
Oh... I just moved the HUD node up before the player node in the parent scene hierarchy, and it works. I'm a bit suspicious though -- does Godot guarantee the ordering of node loading to match the scene tree? Or does it do it in parallel and this could become a race condition?